tests/remote-schema/remote-schema.test.ts

```diff
@@ -0,0 +1,510 @@
+import path from "path";
+import openapiTS from "../../src/index";
+
+describe("remote $refs", () => {
+  it("resolves remote $refs", async () => {
+    const types = await openapiTS(path.join(__dirname, "spec", "spec.yml"));
+    expect(types).toEqual(`/**
+ * This file was auto-generated by openapi-typescript.
+ * Do not make direct changes to the file.
+ */
+
+export interface paths {}
+
+export interface components {
+  schemas: {
+    /** this is a duplicate of subschema/remote1.yml */
+    Circular: string;
+    Remote1: external["subschema/remote1.yml"]["components"]["schemas"]["Remote1"];
+    Pet: external["https://raw.githubusercontent.com/drwpow/openapi-typescript/main/tests/v3/specs/petstore.yaml"]["components"]["schemas"]["Pet"];
+  };
+}
+
+export interface operations {}
+
+export interface external {
+  "https://raw.githubusercontent.com/drwpow/openapi-typescript/main/tests/v3/specs/petstore.yaml": {
```

Comment:
This is the basic format of loading remote schemas: 

```ts
export interface external {
  [namespace]: {
    // … subschema
  }
}
```

Looking at how [json-schema-ref](https://github.com/APIDevTools/json-schema-ref-parser) works, it does a really good job of flattening schemas. But since we’re converting to TypeScript, I have some hesitations in trying to flatten and convert to a different language.

While this output is verbose with the namespacing, it’s not limiting. So I feel it meets one of the project goals: **support any valid OpenAPI schema, no matter how complicated.** The big area of concern is conflicts: just because you reference `User` in a remote schema, doesn’t mean that you want the rest of that remote schema’s definitions to overwrite your own. This approach lets you reference foreign components while still preserving the namespace.

If this is in violation of the spec (e.g. if the spec demands that all names be unique even across remote schemas) I’d be happy to revise it, but I had trouble finding a clear answer on namespacing between remote schemas. To me it seems illogical that a schema would either be valid or invalid based on remote refs, and so preserving local namespacing is preferable.